R/reportFE.R

```diff
@@ -1653,25 +1680,204 @@ reportFE <- function(gdx, regionSubsetList = NULL,
+                 setNames((dimSums(mselect(vm_demFeSector_woNonEn,all_enty1="fegas",all_enty="segafos",emi_sectors="indst", all_emiMkt="ETS") ,dim=3,na.rm=T)), "FE|w/o Non-energy Use|Industry|ETS|Gases|+|Fossil (EJ/yr)"),
+                 setNames((dimSums(mselect(vm_demFeSector_woNonEn,all_enty1="fegas",all_enty="segasyn",emi_sectors="indst", all_emiMkt="ETS") ,dim=3,na.rm=T)), "FE|w/o Non-energy Use|Industry|ETS|Gases|+|Hydrogen (EJ/yr)")
+
+
+    )
+
+      tryCatch(
+        expr = {
+          out <- mbind(
+            out,
+            setNames(
+              out[, , "FE|Industry|Chemicals|+|Solids (EJ/yr)"] - out[, , "FE|Non-energy Use|Industry|+|Solids (EJ/yr)"],
+              "FE|w/o Non-energy Use|Industry|Chemicals|Solids (EJ/yr)"
+            ),
+            setNames(
+              out[, , "FE|Industry|Chemicals|+|Liquids (EJ/yr)"] - out[, , "FE|Non-energy Use|Industry|+|Liquids (EJ/yr)"],
+              "FE|w/o Non-energy Use|Industry|Chemicals|Liquids (EJ/yr)"
+            ),
+            setNames(
+              out[, , "FE|Industry|Chemicals|+|Gases (EJ/yr)"] - out[, , "FE|Non-energy Use|Industry|+|Gases (EJ/yr)"],
+              "FE|w/o Non-energy Use|Industry|Chemicals|Gases (EJ/yr)"
+            ),
+            setNames(
+              out[, , "FE|Industry|+++|Chemicals (EJ/yr)"] - out[, , "FE|Non-energy Use|+|Industry (EJ/yr)"],
+              "FE|w/o Non-energy Use|Industry|Chemicals (EJ/yr)"
+            )
+          )
+        },
+        error = function(e) {
+          warning(e)
+        }
+      )
   }
 
-  # in case the current non-energy use implementation creates negative values, set them to 0
-  if (any(out < 0)) {
-    out[out < 0] <- 0
+  ### FE variables without bunkers ----
+
+  ### variables for which version without bunkers should be calculated
+
+  fe.vars.woBunkers <- c(
+
+    "FE (EJ/yr)",
+    "FE|++|Transport (EJ/yr)",
+    "FE|Transport|+|Liquids (EJ/yr)")
+
+  # add FE w/o non-energy use variables if available
+  if ("FE|Non-energy Use (EJ/yr)" %in% getNames(out)) {
+
+
+    fe.vars.woBunkers <- c(fe.vars.woBunkers,
+                          "FE|w/o Non-energy Use (EJ/yr)",
+                          "FE|w/o Non-energy Use|Liquids (EJ/yr)")
+
+  }
+
+  # bunker correction for distinction of fossil, biomass, hydrogen-based liquids
+  fe.vars.woBunkers.fos <- c(  "FE|Liquids|+|Fossil (EJ/yr)",
+                               "FE|Transport|Liquids|+|Fossil (EJ/yr)")
+
+  fe.vars.woBunkers.bio <- c(  "FE|Liquids|+|Biomass (EJ/yr)",
+                               "FE|Transport|Liquids|+|Biomass (EJ/yr)")
+
+  fe.vars.woBunkers.syn <- c(  "FE|Liquids|+|Hydrogen (EJ/yr)",
+                               "FE|Transport|Liquids|+|Hydrogen (EJ/yr)")
+
+
+  # add FE w/o non-energy use variables if available
+  if ("FE|Non-energy Use (EJ/yr)" %in% getNames(out)) {
+
+    # bunker correction for distinction of fossil, biomass, hydrogen-based liquids
```

Comment:
Shouldn't we also add something here to also get ``FE|w/o Bunkers|w/o Non-energy Use``. So the top-level FE variable without both?